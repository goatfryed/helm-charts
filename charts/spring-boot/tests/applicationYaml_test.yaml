suite: test chart
templates:
  - deployment.yaml
  - propertiesConfigMap.yaml
tests:
  - it: injects application properties spring-container
    values:
      - "values/values.applicationYaml.yaml"
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec.template.spec
      - template: propertiesConfigMap.yaml
        matchSnapshot: {}
  - it: activates profiles automatically
    values:
      - "values/values.applicationYaml.yaml"
    templates:
      - deployment.yaml
    asserts:
      - equal:
          path: spec.template.spec.containers[0].env[0]
          value:
            name: SPRING_PROFILES_ACTIVE
            value: "profile1, profile2, profile3"
  - it: mounts configuration values
    set:
      spring.config.userConfig.values:
        foo: bar
        how:
          are: you
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec.template.spec
  - it: mounts referenced configMaps
    set:
      spring.config.userConfig.configMap.name: user-defined-config-map
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec.template.spec