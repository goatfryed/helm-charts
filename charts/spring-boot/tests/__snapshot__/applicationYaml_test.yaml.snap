injects application properties spring-container:
  1: |
    containers:
      - env:
          - name: SPRING_PROFILES_ACTIVE
            value: profile1, profile2, profile3
        image: :1.16.0
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /
            port: http
        name: spring-boot
        ports:
          - containerPort: 80
            name: http
            protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: http
        resources: {}
        securityContext: {}
        volumeMounts:
          - mountPath: /app/config/helm-config
            name: helm-application-properties
          - mountPath: /app/config/config-map/application-profile3.properties
            name: user-application-properties-profile3
            subPath: my-key
    securityContext: {}
    serviceAccountName: RELEASE-NAME-spring-boot
    volumes:
      - configMap:
          name: RELEASE-NAME-spring-boot-application-properties
        name: helm-application-properties
      - configMap:
          name: user-defined-config-map
        name: user-application-properties-profile3
  2: |
    apiVersion: v1
    data:
      application-profile1.yaml: |
        server:
          port: 8080
      application-profile2.yaml: |
        logging:
          level:
            org.springframework: error
            com.example.myapp: debug
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME-spring-boot-application-properties
mounts configuration values:
  1: |
    containers:
      - env:
          - name: SPRING_PROFILES_ACTIVE
            value: userConfig
        image: :1.16.0
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /
            port: http
        name: spring-boot
        ports:
          - containerPort: 80
            name: http
            protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: http
        resources: {}
        securityContext: {}
        volumeMounts:
          - mountPath: /app/config/helm-config
            name: helm-application-properties
    securityContext: {}
    serviceAccountName: RELEASE-NAME-spring-boot
    volumes:
      - configMap:
          name: RELEASE-NAME-spring-boot-application-properties
        name: helm-application-properties
mounts referenced configMaps:
  1: |
    containers:
      - env:
          - name: SPRING_PROFILES_ACTIVE
            value: userConfig
        image: :1.16.0
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /
            port: http
        name: spring-boot
        ports:
          - containerPort: 80
            name: http
            protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: http
        resources: {}
        securityContext: {}
        volumeMounts:
          - mountPath: /app/config/helm-config
            name: helm-application-properties
          - mountPath: /app/config/config-map/application-userConfig.yaml
            name: user-application-properties-userConfig
            subPath: application.yaml
    securityContext: {}
    serviceAccountName: RELEASE-NAME-spring-boot
    volumes:
      - configMap:
          name: RELEASE-NAME-spring-boot-application-properties
        name: helm-application-properties
      - configMap:
          name: user-defined-config-map
        name: user-application-properties-userConfig
